src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,153 @@
+import {MeetingsAdapter} from '@webex/component-adapter-interfaces';
+import {Observable, fromEvent, concat} from 'rxjs';
+import {filter, finalize, map, publishReplay, refCount} from 'rxjs/operators';
+
+const LOCAL = 'local';
+const MEDIA_READY = 'media:ready';
+
+/**
+ * The `MeetingsSDKAdapter` is an implementation of the `MeetingsAdapter` interface.
+ * This adapter utilizes the Webex JS SDK to create and join webex meetings.
+ *
+ * @class MeetingsSDKAdapter
+ * @extends {MeetingsAdapter}
+ */
+export default class MeetingsSDKAdapter extends MeetingsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.meetingsObservables = {};
+  }
+
+  /**
+   * Register the SDK meeting plugin to the device
+   * and sync the meeting collection from the server.
+   */
+  async connect() {
+    await this.datasource.meetings.register();
+    await this.datasource.meetings.syncMeetings();
+  }
+
+  /**
+   * Unregister the sdk meeting plugin from the device.
+   */
+  async disconnect() {
+    await this.datasource.meetings.unregister();
+  }
+
+  /**
+   * Retrieve the local device media and add them with
+   * the given default media settings to the meeting.
+   *
+   * @param {string} meetingID  meeting ID to retrieve the SDK meeting object to add local media to
+   */
+  async addLocalMedia(meetingID) {
+    const meeting = this.datasource.meetings.getMeetingByType('id', meetingID);
+
+    // default media setting
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+    const [localStream, localShare] = await meeting.getMediaStreams(mediaSettings);
+
+    return meeting.addMedia({
+      localShare,
+      localStream,
+      mediaSettings,
+    });
+  }
+
+  /**
+   * Returns a meeting object with media attached based on a given event type.
+   *
+   * @param {Meeting} meeting outdated meeting object
+   * @param {Object} media   new media stream to attach to the meeting object based on a given event type
```

Comment:
These need alignment. I think @lalli-flores was looking into jsdoc eslint rules:

```suggestion
   * @param {Meeting} meeting Outdated meeting object.
   * @param {Object}    media    New media stream to attach to the meeting object based on a given event type.
```

Reply:
Let's also document the `media.type` and media.stream` props of the media object.